cppcache/src/PdxRemoteWriter.cpp

```diff
@@ -49,18 +49,18 @@ PdxRemoteWriter::PdxRemoteWriter(
 }
 
 PdxRemoteWriter::PdxRemoteWriter(
-    DataOutput& output, std::string pdxClassName,
+    DataOutput& output, std::shared_ptr<PdxType> pdxType,
     std::shared_ptr<PdxTypeRegistry> pdxTypeRegistry)
-    : PdxLocalWriter(output, nullptr, pdxClassName, pdxTypeRegistry),
+    : PdxLocalWriter(output, pdxType, pdxTypeRegistry),
       m_preserveDataIdx(0),
       m_currentDataIdx(-1),
       m_remoteTolocalMapLength(0) {
-  m_preserveData = nullptr;
   if (m_pdxType != nullptr) {
```

Comment:
If `pdxType` is not nullable then this check is unnecessary.

Reply:
Thing is I did leave that because I thought what if pdxType passed were nullptr, but that's not the way it's intended to be used, so I guess it would be best to force a coredump if pdxType is nullptr, right?

Reply:
Yeah, if it is `nullptr` then line 63 is going to fault anyway. Also, if passed here as `nulltpr` but later use excepted `m_pdxType` to not be `nullptr`... boom.